Orange/widgets/visualize/tests/test_owpythagorastree.py

```diff
@@ -73,3 +83,90 @@ def test_compute_center_with_complex_square_angle_with_base_angle(self):
         expected_point = Point(1.43, 3.98)
         self.assertAlmostEqual(point.x, expected_point.x, places=1)
         self.assertAlmostEqual(point.y, expected_point.y, places=1)
+
+
+class TestOWPythagorasTree(WidgetTest):
+    def setUp(self):
+        self.widget = self.create_widget(OWPythagorasTree)
+
+        # Any widget extending `BaseLearner` automatically adds the instances
+        # to the classifier
+        titanic_data = Table('titanic')
+        self.titanic = TreeLearner(max_depth=2)(titanic_data)
+        self.titanic.instances = titanic_data
+
+        housing_data = Table('housing')
+        self.housing = TreeRegressionLearner(max_depth=2)(housing_data)
+        self.housing.instances = housing_data
+
+    def test_input_classification_tree(self):
+        # Tree model must be none before assignment
+        self.assertIsNone(self.widget.model)
+        self.assertIsNone(self.widget.tree_adapter)
+
+        # Test classification tree assingment
+        self.send_signal('Tree', self.titanic)
+        self.assertIsNotNone(self.widget.model)
+        self.assertEqual(self.widget.tree_type, self.widget.CLASSIFICATION)
+        self.assertIsNotNone(self.widget.tree_adapter)
+
+    def test_input_regression_tree(self):
+        # Tree model must be none before assignment
+        self.assertIsNone(self.widget.model)
+        self.assertIsNone(self.widget.tree_adapter)
+
+        # Test classification tree assingment
+        self.send_signal('Tree', self.housing)
+        self.assertIsNotNone(self.widget.model)
+        self.assertEqual(self.widget.tree_type, self.widget.REGRESSION)
+        self.assertIsNotNone(self.widget.tree_adapter)
```

Comment:
These kinds of tests seem awfully implementation dependent and, I may be mistaken, totally useless.

Instead, rather check that for sending in a tree model of depth 1, there are at most three SquareGraphicsItems in the view etc.